src/messaging/ExchangeMgr.cpp

```diff
@@ -203,23 +241,41 @@ void ExchangeManager::DispatchMessage(const PacketHeader & packetHeader, const P
             }
         }
     }
-    // Discard the message if it isn't marked as being sent by an initiator.
-    else
+    // Discard the message if it isn't marked as being sent by an initiator and the message does not needs to send
+    // ack to the peer.
+    else if (!payloadHeader.IsAckMsg())
```

Comment:
I'm not sure I follow.  `IsAckMsg` is checking whether the message we are dispatching is itself an ack.  The comment makes it sound like it wants to check whether the message _needs_ an ack, which would be `IsNeedsAck()` (modulo the renaming I think we should do on that).

It seems to me that either the comment is wrong or this line of code.  Given the code below, I think this line of code is wrong.  Please add a test that would have caught this, if we have any way of testing this code.  If not, please file an issue to add such a test.

Reply:
Good catch! the code is wrong. This needs to be caught by the end- to-end testing. I re-use echo response to ack echo request, so both IsAckMsg and IsNeedsAck are set, that is why I didn't catch this issue.